src/ghci/stdout.rs

```diff
@@ -99,23 +117,16 @@ impl GhciStdout {
             .reader
             .read_until(&bootup_patterns, WriteBehavior::Write, &mut self.buffer)
             .await?;
-        tracing::debug!(?lines, "ghci started");
+        tracing::debug!(?lines, "ghci started, saw version marker");
 
-        let init_prompt_patterns = AhoCorasick::from_anchored_patterns(["ghci> "]);
         // We know that we'll get _one_ `ghci> ` prompt on startup.
-        self.prompt(when_ready, Some(&init_prompt_patterns)).await?;
-        // But we might get more than one `ghci> ` prompt, so digest any others from the buffer.
-        while let Some(lines) = self
-            .reader
-            .try_read_until(
-                &init_prompt_patterns,
-                WriteBehavior::NoFinalLine,
-                &mut self.buffer,
-            )
-            .await?
-        {
-            tracing::debug!(?lines, "initial ghci prompt");
-        }
```

Comment:
Turns out `try_read_until` would _sometimes_ do a blocking read call and get stuck forever here.